crates/nu-utils/standard_library/tests.nu

```diff
@@ -63,8 +63,76 @@ def test_path_add [] {
     }
 }
 
+
+def test_dirs [] {
+
+    def "myassert eq" [
+        left:any 
+        right:any 
+        msg?:string = "..."
+        --verbose:bool = false (-v)  # enable to see successful tests and values of unequal tests
+    ] {
+        if $left != $right {
+            let start_span = (metadata $left).span.start
+            let end_span = (metadata $right).span.end
+            let fail_msg = (if $verbose {
+                                $"\n    left: ($left|debug)\nnot equal to\n   right: ($right|debug)"
+                            } else {""})
+            error make {msg: $"Assertion failed checking ($msg)($fail_msg)", label: {text: "Values not equal" start: $start_span end: $end_span}}
+        } else {
+            if $verbose {
+                echo $"check succeeded: ($msg)"
+            }
+        }
+    }
+    
+    # need some directories to play with
+    let base_path = ($"tmp_(random uuid)" | path expand )
+
+    try {
+        mkdir $base_path ($base_path | path join "a") ($base_path | path join "b")
+        cd $base_path
+        use dirs.nu
+
+        myassert eq 1 ($env.DIRS_LIST | length) "list is just pwd after initialization"
+        myassert eq $base_path $env.DIRS_LIST.0 "list is just pwd after initialization"
+
+        dirs next
+        myassert eq $base_path $env.DIRS_LIST.0 "next wraps at end of list"
+
+        dirs prev
+        myassert eq $base_path $env.DIRS_LIST.0 "prev wraps at top of list"
+
+        dirs add ($base_path | path join "b") ($base_path | path join "a")
+        myassert eq ($base_path | path join "b") $env.PWD "add changes PWD to first added dir"
+        myassert eq 3 ($env.DIRS_LIST | length) "add in fact adds to list"
+        myassert eq ($base_path | path join "a") $env.DIRS_LIST.2 "add in fact adds to list"
+        
+        dirs next 2
+        myassert eq $base_path $env.PWD "next wraps at end of list"
+
+        dirs prev 1 
+        myassert eq ($base_path | path join "a") $env.PWD "prev wraps at start of list"
+
+        dirs drop
+        myassert eq 2 ($env.DIRS_LIST | length) "drop removes from list"    
+        myassert eq $base_path $env.PWD "drop changes PWD to next in list (after dropped element)"
+
+        let show_tab = (dirs show)
+        myassert eq $show_tab [[active path]; [true $base_path] [false ($base_path | path join "b")]] "show table contains expected information"
+    } catch { |error|
```

Comment:
about that `catch` and the `rm` that does not get executed when a test fails :thinking: 
maybe putting the test path in `/tmp/` could solve the issue in that you wouldn't even need to clean it :+1: 

you can use `$nu.temp-path` for cross-platform support :ok_hand:

Reply:
Ah! I didn't know about `$nu.temp-path`, that's good.  
I do think it's bad joss for a test harness to not clean up after itself (at least, to make a good-faith effort), would rather clean up.  E.g Windows doesn't have as active a tmp cleanup job.

Reply:
no problem, that's a good argument :+1: